sycl/source/detail/scheduler/graph_processor.cpp

```diff
@@ -25,8 +25,10 @@ std::vector<EventImplPtr>
 Scheduler::GraphProcessor::getWaitList(EventImplPtr Event) {
   std::vector<EventImplPtr> Result;
   Command *Cmd = getCommand(Event);
-  for (const DepDesc &Dep : Cmd->MDeps)
-    Result.push_back(Dep.MDepCommand->getEvent());
+  for (const DepDesc &Dep : Cmd->MDeps) {
+    if (Dep.MDepCommand)
```

Comment:
Looks like something not directly related to the removing old scheduler, which is supposed to be non-functional change.
Is a bug fix? If so, it would be great to have a regression test.

Reply:
@bader Yes, you are right, it is a bug fix. Let's I will add the regression test.

Reply:
Added test.